external/avro/src/main/scala/org/apache/spark/sql/avro/AvroFileFormat.scala

```diff
@@ -276,10 +274,15 @@ private[avro] object AvroFileFormat {
     }
   }
 
-  def ignoreFilesWithoutExtensions(conf: Configuration): Boolean = {
-    // Files without .avro extensions are not ignored by default
-    val defaultValue = false
+  def ignoreExtension(conf: Configuration, options: Map[String, String]): Boolean = {
+    val ignoreFilesWithoutExtensionByDefault = false
+    val ignoreFilesWithoutExtension = conf.getBoolean(
+      AvroFileFormat.IgnoreFilesWithoutExtensionProperty,
+      ignoreFilesWithoutExtensionByDefault)
 
-    conf.getBoolean(AvroFileFormat.IgnoreFilesWithoutExtensionProperty, defaultValue)
+    options
+      .get("ignoreExtension")
```

Comment:
Shall we document this option somewhere?

Reply:
Sure, I am going to add the `AvroOption` class and document all Avro options there.

Reply:
Let's make sure we describe that in a public API later.